test/unit/sql/index-hints-fragment.test.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const Support   = require('../support');
+
+const current   = Support.sequelize;
+const expectsql = Support.expectsql;
+const sql = current.dialect.queryGenerator;
+const { IndexHints } = require('@sequelize/core');
+
+if (current.dialect.supports.indexHints) {
+  describe(Support.getTestDialectTeaser('SQL'), () => {
+    describe('indexHintsFragment', () => {
+      it('correctly formats the index hints', () => {
+        expectsql(sql.indexHintsFragment([{ type: IndexHints.USE, values: ['index_project_on_name'] }]), {
+          default: 'USE INDEX (`index_project_on_name`)',
```

Comment:
`default` should be written using sql server's identifier quoting syntax (`[index_project_on_name]`) as our test suite will automatically translate it to corresponding syntax for that dialect.

You could remove the snowflake variant, and specify this instead:

```js
default: 'USE INDEX ([index_project_on_name])',
```

and it will be interpreted as this in mysql:

```js
default: 'USE INDEX (`index_project_on_name`)',
```

and this in snowflake:

```js
default: 'USE INDEX ("index_project_on_name")',
```